### PR TITLE
Refactoring/reorganizing tests

### PR DESCRIPTION
**Pull request by drewpts**
[Review on Reviewable](https://reviewable.io/reviews/libproject/libproject/84)

### CODE REVIEW

File: test/libprojecttest.cpp [hunk 1]
```diff
@@ -33,370 +33,185 @@ list<FileSetLoader::Path> pathsToSingleAbnormal = {
...
-
         TestSkeleton::SetUp();
     }
-
     void TearDown() {
-      ofstream out (pathToMainFile);
-      for (const auto& sym : contentBackup)
-        out << sym;
-      out.close();
+        TestSkeleton::TearDown();
```
Comment (mkwork) [inline test/libprojecttest.cpp:146]: _#bug_
why do you need those rows? Them are do nothing

File: test/libprojecttest.cpp [hunk 2]
```diff
@@ -33,370 +33,185 @@ list<FileSetLoader::Path> pathsToSingleAbnormal = {
...
-        buf = 0;
-        while(in.get(buf))
-          os_case_removing_lone_subproject << buf;
-        in.close();
-        contentBackup_withoutSubprojects = os_case_removing_lone_subproject.str();
-
         TestSkeleton::SetUp();
     }
     void TearDown() {
+        TestSkeleton::TearDown();
```
Comment (mkwork) [inline test/libprojecttest.cpp:171]: **#bug**
stale dead code

**Comment by mkwork** [comment]
Review status: 0 of 7 files reviewed, 3 unresolved discussions, all commit checks successful.

---

<sup>**[test/libprojecttest.cpp, line 115 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqRN8XBHEsQTRQb2IX9)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L115)):</sup>
what about using an std::unique_ptr ?

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/libproject/libproject/84)

**Comment by mkwork** [comment]
Review status: 0 of 7 files reviewed, 4 unresolved discussions, all commit checks successful.

---

<sup>**[test/libprojecttest.cpp, line 134 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqRNP0FRFfgvIo0C_SJ)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L134)):</sup>
code does nothing

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/libproject/libproject/84)

**Comment by mkwork** [comment]
Review status: 0 of 7 files reviewed, 4 unresolved discussions, all commit checks successful.

---

<sup>**[test/libprojecttest.cpp, line 134 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqRNhecHu-8wO92Z9hy)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L134)):</sup>
code does nothing

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/libproject/libproject/84)

**Comment by mkwork** [comment]
Review status: 0 of 7 files reviewed, 7 unresolved discussions, all commit checks successful.

---

<sup>**[test/libprojecttest.cpp, line 289 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqRO3fBPZizteHgUfZQ)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L289)):</sup>
code does nothing again...

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/libproject/libproject/84)

**Comment by mkwork** [comment]
Review status: 0 of 7 files reviewed, 8 unresolved discussions, all commit checks successful.

---

<sup>**[test/libprojecttest.cpp, line 307 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqROC2WZPUCMvyqK9CZ)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L307)):</sup>
why `map<string, FileSetLoader*>` isn't typedefed to something like `FilesetLoader::Map` ?

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/libproject/libproject/84)

**Comment by mkwork** [comment]
Review status: 0 of 7 files reviewed, 9 unresolved discussions, all commit checks successful.

---

<sup>**[test/libprojecttest.cpp, line 346 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqRORN0Wnp5PxE4-Tnt)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L346)):</sup>
you already know what i'll say :)
_code does nothing_

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/libproject/libproject/84)

**Comment by mkwork** [comment]
Review status: 0 of 7 files reviewed, 10 unresolved discussions, all commit checks successful.

---

<sup>**[test/libprojecttest.cpp, line 349 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqROYpXgTuVdRDkNgID)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L349)):</sup>
code does nothing...

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/libproject/libproject/84)

**Comment by mkwork** [comment]
Review status: 2 of 7 files reviewed, 11 unresolved discussions, all commit checks successful.

---

<sup>**[test/libprojecttest.cpp, line 795 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqROvU76Mcbh1pe6lL1)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L795)):</sup>
Gmock used, but gtest initialized

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/libproject/libproject/84)

**Comment by mkwork** [comment]
Review status: all files reviewed, 11 unresolved discussions, all commit checks successful.
Reviewed files:
- src/libproject.cpp @ r1
- test/libprojecttest.cpp @ r1
- test/project_files/getsubprojectspaths/getpath.libproject @ r1
- test/project_files/getsubprojectspaths/getpaths.libproject @ r1
- test/project_files/getsubprojectspaths/nosubprojects.libproject @ r1
- test/project_files/getsubprojectspaths/sub/s1.libproject @ r1
- test/project_files/getsubprojectspaths/sub/s2.libproject @ r1

---

<sup>**[test/libprojecttest.cpp, line 168 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqRNrdIpoylC19nwfdO)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L168)):</sup>
code does nothing

---

<sup>**[test/libprojecttest.cpp, line 192 [r1]](https://reviewable.io:443/reviews/libproject/libproject/84#-JqRNzc9YjZTCQmn1H8r)** ([raw file](https://github.com/libproject/libproject/blob/8d59546dea69e1f8c349fb2e0667ea14f18c13c1/test/libprojecttest.cpp#L192)):</sup>
code does nothing

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/libproject/libproject/84)

File: test/libprojecttest.cpp (hunk 1 above)
Comment (drewpts) [inline test/libprojecttest.cpp:146]: @mkwork 
In TestSkeleton::TearDown and SetUp I put things that performs some work. Why "do nothing"?

File: test/libprojecttest.cpp (hunk 2 above)
Comment (drewpts) [inline test/libprojecttest.cpp:171]: @mkwork 
:weary:

**Comment by drewpts** [comment]
@mkwork 
Why you chose Reviewable.io? :) I can't answer direct to your comments: ok, I'll use unique_ptr and typedef to std::map.

**Comment by drewpts** [comment]
Closing request. I need to fix issues.
- [x] use unique_ptr
- [x] typedef std::map
- [x] ooh, stale dead code. " Hmm, maybe @mkwork understood it all wrong?... "

**Comment by mkwork** [comment]
in a reference for [this comment](https://github.com/libproject/libproject/pull/84#issuecomment-106704125)

> Hmm, maybe @mkwork understood it all wrong?

Your SetUp and TearDown in style of:

``` cpp
struct Derived : BaseClass {
  void SetUp {
     BaseClass::SetUp();
  }
};
```

are doing nothing, except of taking place

**Comment by drewpts** [comment]
@mkwork 
Thanks for your explanation. I just little bit confused for this shame. :) Of course you're right!

**Pull request closed without merging**